fix: correctly detect DROP COLUMN vs DROP TABLE in migration analyzer

webjunkie opened the pull request:
## Summary

Fixed migration analyzer bug where `ALTER TABLE ... DROP COLUMN IF EXISTS` was incorrectly classified as `DROP TABLE`, and aligned DROP COLUMN validation with DROP TABLE pattern.

## Changes

### Bug Fix
- Fixed keyword matching order in RunSQLAnalyzer - check for "COLUMN" before "TABLE"
- `ALTER TABLE foo DROP COLUMN IF EXISTS bar` now correctly identified as DROP COLUMN

### Alignment with DROP TABLE Pattern
- Properly staged column drops (with prior SeparateDatabaseAndState + RemoveField) get score 2 (NEEDS_REVIEW) instead of always score 5 (BLOCKED)
- Refactored `check_drop_table_properly_staged()` to `check_drop_properly_staged()` supporting both tables and columns
- Added regex parsing to extract table and column names from SQL
- Enhanced model name extraction for multi-word app labels (e.g., `llm_analytics`)

### Test Coverage
- `test_run_sql_alter_table_drop_column_if_exists` - DROP COLUMN not confused with DROP TABLE
- `test_run_sql_drop_table_not_confused_with_drop_column` - DROP TABLE still works
- `test_drop_column_with_prior_state_removal` - properly staged drops get score 2
- All 63 tests passing

## Validation

Properly staged column drop (example from PR #40533):
```python
# Migration 0006: Remove from state
migrations.SeparateDatabaseAndState(
    state_operations=[migrations.RemoveField(model_name="Evaluation", name="prompt")],
    database_operations=[],
)

# Migration 0007: Drop column (now gets score 2 instead of 5)
migrations.RunSQL("ALTER TABLE llm_analytics_evaluation DROP COLUMN IF EXISTS prompt;")
```

## Test Plan
- [x] All existing tests pass (63/63)
- [x] Added 3 new tests for DROP COLUMN detection
- [x] Verified DROP TABLE validation still works correctly
- [x] Tested multi-word app label extraction

webjunkie committed:
fix: correctly detect DROP COLUMN vs DROP TABLE in migration analyzer

Migration analyzer was incorrectly classifying `ALTER TABLE ... DROP COLUMN IF EXISTS`
as `DROP TABLE` because it checked keywords in the wrong order (both contain "DROP", "TABLE",
and "IF EXISTS").

Root cause:
- Keyword matching checked "DROP" + "TABLE" + "IF EXISTS" before checking for "COLUMN"
- SQL like `ALTER TABLE foo DROP COLUMN IF EXISTS bar` matched table drop pattern

Changes:
- Check for "COLUMN" keyword before "TABLE" in DROP operations
- Align DROP COLUMN validation with DROP TABLE pattern:
  - Properly staged column drops (prior SeparateDatabaseAndState with RemoveField) → score 2
  - Unstaged column drops → score 5
- Refactor `check_drop_table_properly_staged()` to `check_drop_properly_staged()` supporting both tables and columns
- Fix model name extraction for multi-word app labels (e.g., `llm_analytics_evaluation` → `Evaluation`)

Tests:
- Added test for DROP COLUMN with prior state removal (score 2)
- Added test ensuring DROP COLUMN not confused with DROP TABLE
- Added test ensuring DROP TABLE still works correctly
- All 63 tests pass

Fixes the issue reported in PR #40533 where properly staged column drops were incorrectly blocked.

webjunkie committed:
refactor: remove backward compatibility wrapper

No external users, direct call to check_drop_properly_staged is cleaner.

carlos-marchal-ph reviewed (approved):
LGTM, thanks for the quick fix!

webjunkie committed:
Merge remote-tracking branch 'origin/master' into fix/migration-analyzer-drop-column-detection

webjunkie committed:
fix: prevent false positive when table name contains 'column'

Move regex match check before entering DROP COLUMN branch to avoid
misclassifying 'DROP TABLE column_data' as DROP COLUMN operation.

Addresses review feedback from @greptile-apps[bot]
